config/src/network_id.rs

```diff
@@ -81,12 +81,66 @@ impl NetworkContext {
 /// and handshakes should verify that the NetworkId being used is the same during a handshake,
 /// to effectively ensure communication is restricted to a network.  Network should be checked that
 /// it is not the `DEFAULT_NETWORK`
-#[derive(Clone, Deserialize, Eq, Hash, PartialEq, Serialize)]
-#[serde(rename = "NetworkId", rename_all = "snake_case")]
+#[derive(Clone, Copy, Eq, Hash, PartialEq)]
+#[repr(u8)]
 pub enum NetworkId {
-    Validator,
+    Validator = 0,
     Public,
-    Private(String),
+    Vfn,
```

Comment:
Oh I think you need to explicitly set the values here with the gap, otherwise it won't be compatible

```rust
#[repr(u8)]
pub enum NetworkId {
    Validator = 0,
    Public = 1,
    Vfn = 3,
}
```

Reply:
Since we're using a conversion layer, we can make it whatever we want, so I've reordered it with no gap.